aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -952,25 +955,33 @@ Tensor diff(const Tensor& self, int64_t n, int64_t dim, const c10::optional<Tens
 
 static inline Tensor& diff_out_helper(const Tensor& self, int64_t n, int64_t dim, Tensor& result) {
   if (n == 0) {
-    at::native::resize_output(result, self.sizes());
+    if (resize_output_check_symint(result, self.sym_sizes())) {
```

Comment:
I see there's also an `at::native::resize_output::symint()`, does that not work here?

Reply:
That is a good question. I looked into this a little more.

We should not call the `at::native::resize_output_symint` directly because that impl is specific to the meta kernel, it will not reallocate any memory for you underneath. Something like `at::resize__symint` works because dispatcher decides whether we actually call into `at::native::resize__symint` vs `at::native::resize_`.

The problem with `at::resize__symint` is that since it is not out-specific anymore, we now have to produce the "you resized a non-empty tensor" warning ourselves. We would prefer to analogously call `resize_output_symint` through the dispatcher. There are two issues with that (1) at::_resize_output_  is not symintified yet and (2) even if we symintify _resize_output_, compilation still won't work because _resize_output_ is not supported by FakeTensor because its part of `_device_not_kwarg_ops`.

Something that I haven't looked into yet and what I'm filing an issue for is the case where even if we do use `resize__symint`, there is still a case where we error.

Reply:
Added a comment to `at::native::resize_output_symint` to warn people not to use this directly.

> Something that I haven't looked into yet and what I'm filing an issue for is the case where even if we do use resize__symint, there is still a case where we error.

Filed an issue https://github.com/pytorch/pytorch/issues/111580